src/cwrapper.cpp

```diff
@@ -23,127 +23,172 @@ using CSymPy::Number;
 using CSymPy::rcp_static_cast;
 using CSymPy::is_a;
 
-struct CWrapper {
-    RCP<const Basic> ptr;
-};
+#define RCP_cast(x) (static_cast<RCP<const Basic> *>(static_cast<void *>(x)));
```

Comment:
Ha, yes! Thanks for noticing this. I couldn't figure it out. I just pushed in a patch that uses this and things start to look much better.